File: atcoder/contests/abc388/d.c
#include <stdio.h>

int main(){

    int N, i, aliens[100005];

    scanf(" %d", &N);

    for (i = 0; i < N; i++){
        scanf(" %d", &aliens[i]);
    }

    for (i = 0; i < N; i++){
        for (int k = 0; k < i; k++){
            if (aliens[k] > 0){
                aliens[k]--;
                aliens[i]++;
            }
        }
    }

    printf("%d", aliens[0]);
    for (i = 1; i < N; i++){
        printf(" %d", aliens[i]);
    }
    printf("\n");
    return 0;
}

File: atcoder/contests/abc388/d.py
"""
Em um certo planeta, existem N aliens, todos menores de idade.

O i-th alien atualmente tem Ai pedras, e consegue ser adulto exatamente i anos depois.

Quando alguem vira adulto nesse planeta, todo adulto que tem pelo menos uma pedra dá exatamnte 
uma pedra como presente de felicitação ao aliem que acabou de se tornar adulto.

Descubra quantas pedras cada alienígena terá depois de N anos.

Suponha que nenhum novo aliem nascerá no futuro
""" 

N = int(input())

aliens = list(map(int, input().split()))

delta = [0] * (N + 1)

give_stones = 0

for i in range(N):
    give_stones += delta[i]
    aliens[i] += give_stones

    end = min(N - 1, i + aliens[i])

    aliens[i] -= end - i

    delta[i] += 1
    delta[end + 1] -= 1
    give_stones += 1

print(" ".join(map(str, aliens)))
